fat/fat2/pticker.go

```diff
@@ -0,0 +1,111 @@
+package fat2
+
+import "fmt"
+
+// PPTicker is an internal representation of a PegNet asset type
+type PTicker int
+
+const (
+	PTickerInvalid PTicker = iota
+	PTickerPEG
+	PTickerUSD
+	PTickerEUR
+	PTickerJPY
+	PTickerGBP
+	PTickerCAD
+	PTickerCHF
+	PTickerINR
+	PTickerSGD
+	PTickerCNY
+	PTickerHKD
+	PTickerKRW
+	PTickerBRL
+	PTickerPHP
+	PTickerMXN
+	PTickerXAU
+	PTickerXAG
+	PTickerXBT
+	PTickerETH
+	PTickerLTC
+	PTickerRVN
+	PTickerXBC
+	PTickerFCT
+	PTickerBNB
+	PTickerXLM
+	PTickerADA
+	PTickerXMR
+	PTickerDAS
+	PTickerZEC
+	PTickerDCR
+	PTickerMax
+)
+
+// TODO: figure out how to handle "tUSD" and other testnet types
+//       or convince PegNet core devs that the differentiation is unnecessary
+//       due to the existence of multiple testnets
+var validPTickers = map[string]PTicker{
```

Comment:
We might as well do this right to begin with and it will solve the Marshaling inefficiency below.

We need an array for marshalling and and a map for unmarshaling. Let's populate the array statically, and use an initialization function for the map.

Something like:
```
var pTickerStrings = [...]string {
	"pPEG",
	"pUSD",
...
} 
```
Of course the appropriate bounds checking and offset need to be applied to the PTicker value inside of Marshal.

Then init the map like so:
```
var validPTickers = func() map[string]PTicker {
   for i, str := range pTickerStrings {
   ...
   }
}()
```

Reply:
Really like the way this looks now. Done